bazel/protobuf.bzl

```diff
@@ -146,12 +124,48 @@ cc_proto_gen_validate = rule(
     implementation = _protoc_gen_validate_cc_impl,
 )
 
-java_proto_gen_validate = rule(
-    attrs = {
-        "deps": attr.label_list(
-            mandatory = True,
-            providers = ["proto"],
+_ProtoValidateSourceInfo = provider(
+    fields = {
+        "sources": "Depset of sources created by protoc with protoc-gen-validate plugin",
+    },
+)
+
+def _create_include_path(include):
+    return "--proto_path={0}={1}".format(_proto_path(include), include.path)
+
+def _proto_validate_aspect_impl(target, ctx):
```

Comment:
Can you rename this so it's obvious that it's Java-specific? We'll likely want to do this for C++ once Bazel supports C++ compilation helpers (like `java_common` for Java).

Reply:
The name is still very general though the implementation is Java-specific.

Reply:
Sorry about that, pushed an incomplete commit